luigi/target.py

```diff
@@ -336,6 +341,8 @@ def move_to_final_destination(self):
         raise NotImplementedError()
 
     def __del__(self):
+        self.unintended_close = True
+        self.close()
```

Comment:
This pattern is hard to read I think. If `close()` depends on the attribute `unintended_close` it make much more sense for it to be a parameter to `close()`